### PR TITLE
fix some typos

### PR DESCRIPTION
**Pull request by zhenghenghuo**
Signed-off-by: zhenghenghuo <zhenghenghuo@zju.edu.cn>

**- What I did**
fix some typos

**- How I did it**
"an" should be used before some words such as "STS" and "HTTP", and I change these "a" to "a"

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]
Please do not modify vendored files

**Comment by zhenghenghuo** [comment]
but I don't know which kind of files are vendored files?

**Comment by AkihiroSuda** [comment]
The files under the `vendor` directory.

These files are synced with other git repositories, using [vndr](https://github.com/LK4D4/vndr).
Please also refer to `vendor.conf`: https://github.com/docker/docker/blob/master/vendor.conf.

**Comment by zhenghenghuo** [comment]
Thanks, and I remove those changes in vendored files.

**Review by AkihiroSuda — APPROVED** [review]
LGTM, but just note that some people pronounce SSH as /ssh/, SHA as /shaw/ :smile: 

http://www.linuxquestions.org/questions/general-10/how-do-you-pronounce-ssh-637508/
https://en.wikipedia.org/wiki/Talk:SHA-1/Archive_1

**Comment by zhenghenghuo** [comment]
Thanks.

File: docs/reference/commandline/pull.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ debian       jessie   f50f9524513f    5 days ago   125.1 MB
 debian       latest   f50f9524513f    5 days ago   125.1 MB
 ```
 
-Docker uses a content-addressable image store, and the image ID is a SHA256
+Docker uses a content-addressable image store, and the image ID is an SHA256
```
Comment (thaJeztah) [inline docs/reference/commandline/pull.md]: agree with @AkihiroSuda I think this should be `a SHA256`

File: image/spec/v1.1.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ This specification uses the following terms:
     </dt>
     <dd>
         Layers are referenced by cryptographic hashes of their serialized
-        representation. This is a SHA256 digest over the tar archive used to
+        representation. This is an SHA256 digest over the tar archive used to
```
Comment (thaJeztah) [inline image/spec/v1.1.md]: same here

File: image/spec/v1.2.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ This specification uses the following terms:
     </dt>
     <dd>
         Layers are referenced by cryptographic hashes of their serialized
-        representation. This is a SHA256 digest over the tar archive used to
+        representation. This is an SHA256 digest over the tar archive used to
```
Comment (thaJeztah) [inline image/spec/v1.2.md]: and here

**Review by thaJeztah — CHANGES_REQUESTED** [review]
left a comment about the `SHA256`; can you change those back? Otherwise looks good

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Comment by thaJeztah** [comment]
Looks like your last push added a merge commit; can you rebase again?

**Comment by thaJeztah** [comment]
If you need help squashing your commits; the contributors guide has a section on rebasing/squashing; https://docs.docker.com/opensource/workflow/work-issue/#/pull-and-rebase-frequently

**Pull request merged**